zerver/lib/events.py

```diff
@@ -201,41 +205,68 @@ def fetch_initial_state_data(user_profile: UserProfile,
+                user_profile,
+                medium=True,
+                client_gravatar=False,
+            )
+            state['avatar_url'] = avatar_url(
+                user_profile,
+                medium=False,
+                client_gravatar=False,
+            )
+
+            state['can_create_streams'] = user_profile.can_create_streams()
+            state['can_subscribe_other_users'] = user_profile.can_subscribe_other_users()
+            state['cross_realm_bots'] = list(get_cross_realm_dicts())
+            state['is_admin'] = user_profile.is_realm_admin
+            state['is_owner'] = user_profile.is_realm_owner
+            state['is_guest'] = user_profile.is_guest
+            state['user_id'] = user_profile.id
+            state['enter_sends'] = user_profile.enter_sends
+            state['email'] = user_profile.email
+            state['delivery_email'] = user_profile.delivery_email
+            state['full_name'] = user_profile.full_name
+        else:
+            state['avatar_source'] = fake_user.avatar_source
+            state['avatar_url_medium'] = avatar_url(
+                fake_user,
+                medium=True,
+                client_gravatar=False,
+            )
+            state['avatar_url'] = avatar_url(
+                fake_user,
+                medium=False,
+                client_gravatar=False,
+            )
 
-        # For the user's own avatar URL, we force
-        # client_gravatar=False, since that saves some unnecessary
-        # client-side code for handing medium-size avatars.  See #8253
-        # for details.
-        state['avatar_source'] = user_profile.avatar_source
-        state['avatar_url_medium'] = avatar_url(
-            user_profile,
-            medium=True,
-            client_gravatar=False,
-        )
-        state['avatar_url'] = avatar_url(
-            user_profile,
-            medium=False,
-            client_gravatar=False,
-        )
-
-        state['can_create_streams'] = user_profile.can_create_streams()
-        state['can_subscribe_other_users'] = user_profile.can_subscribe_other_users()
-        state['cross_realm_bots'] = list(get_cross_realm_dicts())
-        state['is_admin'] = user_profile.is_realm_admin
-        state['is_owner'] = user_profile.is_realm_owner
-        state['is_guest'] = user_profile.is_guest
-        state['user_id'] = user_profile.id
-        state['enter_sends'] = user_profile.enter_sends
-        state['email'] = user_profile.email
-        state['delivery_email'] = user_profile.delivery_email
-        state['full_name'] = user_profile.full_name
+            state['can_create_streams'] = False 
+            state['can_subscribe_other_users'] = False
+            state['cross_realm_bots'] = list(get_cross_realm_dicts())
+            state['is_admin'] = False 
+            state['is_owner'] = False 
+            state['is_guest'] = False 
+            state['user_id'] = fake_user.id
```

Comment:
This is just `None` because a new object doesn't get `id` until being saved to the database

Reply:
Though other attributes are also being used below without being set, so I'm guessing this is intentional after all